### PR TITLE
Response to Diagnostic

### PR DESCRIPTION
**Pull request by dumplingpuff**

### CODE REVIEW

**Comment by dumplingpuff** [comment]
Comfort: 3
Clarity: 3

I used class notes and Mongo DB documentation.

File: diagnostic.js [hunk 1]
```diff
@@ -28,22 +28,47 @@ const mongoose = require('mongoose');
 mongoose.Promise = global.Promise;
 mongoose.connect('mongodb://localhost/mongoose-crud');
 
+const Movement = ('app/models/movement');
```
Comment (GA-MEB) [inline diagnostic.js:31]: Incorrect `require` setup.

File: diagnostic.js [hunk 2]
```diff
@@ -28,22 +28,47 @@ const mongoose = require('mongoose');
...
+
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+// My comments:  This is not complete and I feel like I did the diagnostic
+// based off of Antony's api.
+
+const create = (req, res, next) => {
```
Comment (GA-MEB) [inline diagnostic.js:38]: This isn't Express code - there is no `req` or `res`!

File: models/movement.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+const mongoose = require('mongoose');
+
+const movementSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true
+  },
+  description: String,
```
Comment (GA-MEB) [inline models/movement.js:10]: `description` also needs a `maxlength` validation:

> "A description (no longer than 200 characters)."

File: models/movement.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  startYear: String,
+  endYear: String,
+  _movement: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref:'Movement'
+  }
```
Comment (GA-MEB) [inline models/movement.js:16]: So close! The name for this reference should be `_ancestor`, not `_movement`.

**Comment by GA-MEB** [comment]
So...  your Model setup is pretty on-point, but all of your CRUD actions are wrong because you wrote them as if they were part of an Express app. 

I'd strongly recommend that you resubmit and give this one another shot.

**Pull request closed without merging**